packages/cli/test/integration.js

```diff
@@ -3675,3 +3675,79 @@ test('[vc link] should support the `--project` flag', async t => {
     formatOutput(output)
   );
 });
+
+test('vercel.json projectSettings should be applied to deployment', async t => {
+  const directory = path.join(__dirname, 'fixtures/unit/vercel-json-overrides');
+
+  const deployment = await execa(binaryPath, [
+    directory,
+    ...defaultArgs,
+    '--public',
+    '--confirm',
+  ]);
+
+  t.is(
+    deployment.exitCode,
+    0,
+    formatOutput({
+      stderr: deployment.stderr,
+      stdout: deployment.stdout,
+    })
+  );
+
+  const page = await fetch(deployment.stdout);
+  const text = await page.text();
+  t.is(text, `Hello, World\n`);
+});
+
+test('invalid vercel.json projectSettings should result in a deployment error', async t => {
+  const directory = path.join(
+    __dirname,
+    'fixtures/unit/vercel-json-overrides-error'
+  );
+
+  async function testDeployment() {
+    try {
+      await execa(binaryPath, [
+        directory,
+        ...defaultArgs,
+        '--public',
+        '--confirm',
+      ]);
+      t.fail('Deployment should fail');
+    } catch ({ exitCode, stderr, stdout }) {
```

Comment:
You can do this without try/catch by passing `reject: false` option to execa (see other tests). That will ensure the test fails if the exit code was sucecssful.